File: Algorithm_240_Search a 2D Matrix II.java
// 思路：起始位置设为matrix的右上角。当target > matrix[row][col]时，target一定不在本行中，row++
// 当target < matrix[row][col]时，target一定不在本列中，col--

class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        if (matrix == null || matrix.length == 0 || matrix[0] == null || matrix[0].length == 0) {
            return false;
        }
        int row = 0, col = matrix[0].length - 1;
        while (row < matrix.length && col >= 0) {
            if (target == matrix[row][col]) {
                return false;
            } else if (target > matrix[row][col]) {
                row++;
            } else {
                col--;
            }
        }
        return false;
    }
}

File: Algorithm_240_Search a 2D Matrix II.py
class Solution:
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        if not matrix or not matrix[0]:
            return False
        i, j = 0, len(matrix[0]) - 1
        while j >= 0 and i < len(matrix):
            if matrix[i][j] == target:
                return True
            elif matrix[i][j] < target:
                i += 1
            else:
                j -= 1
        return False